kvbc/include/merkle_processor/SyncMerkleProcessor.h

```diff
@@ -0,0 +1,43 @@
+// Concord
+//
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#pragma once
+
+#include "BaseMerkleProcessor.h"
+
+namespace concord {
+namespace kvbc {
+namespace sparse_merkle {
+
+using concordUtils::bufferToHex;
+
+class SyncMerkleProcessor : public BaseMerkleProcessor {
+ public:
+  virtual ~SyncMerkleProcessor() = default;
+
+  virtual void Init(uint numVersionsStored) override;
+
+  virtual void BeginVersionUpdateBatch() override;
+  virtual void UpdateAccountTree(const address& addr, const std::string& key, const value& data) override;
+
+  virtual void CommitVersionUpdateBatch() override;
+
+  virtual std::vector<std::string> GetAccountMerkleRootPath(const address& addr) override;
+  virtual std::vector<std::string> GetAccountStorageKeyMerklePath(const address& addr, const std::string& key) override;
```

Comment:
Since the return type here is the same as the previous function, there should be some documentation explaining what the difference between the paths is.

Moreover, the dynamic allocations for the path vectors will be expensive (assuming proofs are requested in the hot path), do we have an upper bound for the path length?

After our last discussion I thought that path parts are 4-bit symbols, shouldn't the type reflect that somehow?

Reply:
Those are proof paths which essentially are a list of hashes. Our last discussion was focused on the Nibbles (4 bit symbols), which are a primitive used to address the sub-tree records in the database and will also be used for the generation of the list of hashes here.